Sources/Basics/FileSystem/AsyncFileSystem.swift

```diff
@@ -32,8 +32,8 @@ public actor AsyncFileSystem: Actor {
 
     /// Initialize a new instance of the actor.
     /// - Parameter implementation: an underlying synchronous filesystem.
-    public init(implementation: FileSystem) {
-        self.implementation = implementation
+    public init(_ implementationInitializer: @Sendable () -> FileSystem) {
```

Comment:
This seems to be the only way to initialize `AsyncFileSystem` from a non-`Sendable` type conforming to `FileSystem`.

Reply:
nit: we usually call this "underlying" instead of "implementation"

Reply:
Right, I saw that in `LegacyHTTPClient`, but using an adjective for a variable name felt slightly wrong to me, `underlying` didn't seem to be self-descriptive enough. Would `underlyingImplementation` work, or is that too verbose?